libdebug/cffi/ptrace_cffi_build.py

```diff
@@ -5,10 +5,123 @@
 #
 
 import platform
+from pathlib import Path
 
 from cffi import FFI
 
 if platform.machine() == "x86_64":
+    # We need to determine if we have AVX, AVX2, AVX512, etc.
+    path = Path("/proc/cpuinfo")
+
+    try:
+        with path.open() as f:
+            cpuinfo = f.read()
+    except OSError as e:
+        raise RuntimeError("Cannot read /proc/cpuinfo. Are you running on Linux?") from e
+
+    if "avx512" in cpuinfo:
+        fp_regs_struct = """
+        struct reg_128
+        {
+            unsigned char data[16];
+        };
+
+        struct reg_256
+        {
+            unsigned char data[32];
+        };
+
+        struct reg_512
+        {
+            unsigned char data[64];
+        };
+
+        struct fp_regs_struct
+        {
+            unsigned long type;
+            unsigned char padding0[32];
+            struct reg_128 st[8];
+            struct reg_128 xmm0[16];
+            unsigned char padding1[96];
+            // end of the 512 byte legacy region
+            unsigned char padding2[64];
+            // ymm0 starts at offset 576
+            struct reg_128 ymm0[16];
+            unsigned char padding3[320];
+            // zmm0 starts at offset 1152
+            struct reg_256 zmm0[16];
+            // zmm1 starts at offset 1664
+            struct reg_512 zmm1[16];
+            unsigned char padding4[8];
+        };
+        """
+
+        fpregs_define = """
+        #define FPREGS_AVX 2
+        """
+    elif "avx" in cpuinfo:
+        fp_regs_struct = """
+        struct reg_128
+        {
+            unsigned char data[16];
+        };
+
+        struct reg_256
+        {
+            unsigned char data[32];
+        };
+
+        struct fp_regs_struct
```

Comment:
You should include a reference for this structure for easier future maintenance. Same for the other branches.